app/components/e-heading.vue

```diff
@@ -0,0 +1,181 @@
+          return [
+            'blue',
+            'green',
+            'gray',
+          ].includes(value);
+        },
+      },
+
+      /**
+       * Defines the tagname of the heading (h1-h6)
+       */
+      tagName: {
+        type: String,
+        required: true,
+        validator(value) {
+          return [
+            'h1',
+            'h2',
+            'h3',
+            'h4',
+            'h5',
+            'h6',
+          ].includes(value);
+        },
+      },
+
+      /**
+       * Defines if the heading should have an underline
+       */
+      underline: {
+        type: Boolean,
+        default: true,
+      },
+
+      /**
+       * Defines the font weight
+       */
+      weight: {
+        type: String,
+        default: 'normal',
+        validator(value) {
+          return [
+            'normal',
+            'semibold',
+            'bold',
+          ].includes(value);
+        },
+      },
+    },
+
+    // computed: {},
+    // watch: {},
+
+    // beforeCreate() {},
+    // created() {},
+    // beforeMount() {},
+    // mounted() {},
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    // beforeDestroy() {},
+    // destroyed() {},
+
+    // methods: {},
+
+    /**
+     * Creates a heading element
+     *
+     * @param   {function}    createElement   Vue helper
+     *
+     * @returns {*}
+     */
+    render(createElement) {
+      const element = this.$props.tagName;
+      const attributes = {
+        class: this.b({
+          color: this.$props.color,
+          underline: this.$props.underline,
+          weight: this.$props.weight,
+          tagname: this.$props.tagName,
+        }),
+        attrs: {
+          ...this.$attrs,
+        },
+      };
+
+      return createElement(
+        element,
+        attributes,
+        this.$slots.default,
+      );
+    },
+  };
+</script>
+
+<style lang="scss">
+  .e-heading {
+    font-family: $font-family--primary;
+    font-weight: normal;
+    color: $color-secondary--1;
+    letter-spacing: 2.03px;
+    line-height: 23px;
+    text-decoration: none;
+
+    &--tagname-h1 {
+      font-size: $font-size--18;
+    }
+
+    &--tagname-h2 {
+      font-size: $font-size--16;
+    }
+
+    &--tagname-h3 {
+      font-size: $font-size--14;
+    }
+
+    &--tagname-h4 {
+      font-size: $font-size--14;
+      font-weight: $font-weight--bold;
+    }
+
+    &--tagname-h5 {
+      font-size: $font-size--14;
+      font-weight: $font-weight--semi-bold;
+    }
+
+    &--tagname-h6 {
+      font-size: $font-size--14;
+    }
+
+    &--weight-normal {
+      font-weight: $font-weight--regular;
+    }
+
+    &--weight-semibold {
+      font-weight: $font-weight--semi-bold;
+    }
+
+    &--weight-bold {
+      font-weight: $font-weight--bold;
+    }
+
+    &--underline {
+      display: inline-block;
+      border-bottom: 1px solid $color-primary--1;
+      padding-left: $spacing--20;
```

Comment:
- [ ] Please always use shorthand notation for padding and margin if you define an initial value:

padding: <top> <right> <bottom> <left>
padding: $spacing--15 $spacing--20; (top/bottom left/right)